exonum/src/events/tests.rs

```diff
@@ -45,19 +45,26 @@ impl TestHandler {
         network_requests_tx: mpsc::Sender<NetworkRequest>,
         network_events_rx: mpsc::Receiver<NetworkEvent>,
     ) -> TestHandler {
-        let timer = Timer::default();
-        let receiver = timer.timeout_stream(network_events_rx, Duration::from_secs(30));
         TestHandler {
             handle: None,
             listen_address,
+            network_events_rx: Some(network_events_rx),
             network_requests_tx,
-            network_events_rx: receiver.wait(),
         }
     }
 
     pub fn wait_for_event(&mut self) -> Result<NetworkEvent, ()> {
-        let event = self.network_events_rx.next().unwrap()?;
-        Ok(event)
+        let rx = self.network_events_rx.take().expect("event stream gone");
```

Comment:
It seems you can use [`Stream::by_ref()`](https://docs.rs/futures/0.1.23/futures/stream/trait.Stream.html#method.by_ref) here in order to not bother with wrapping the stream into an `Option<_>`:

```rust
let rx = self.network_events_rx.by_ref();
let future = rx.into_future()
    .deadline(Instant::now() + Duration::from_secs(30))
    .map_err(drop);

let mut core = Core::new().unwrap();
let (event, _) = core.run(future)?;
event.ok_or(())
```

Reply:
Awesome! Thank you.

This also solves the issue with retrieving the stream in case of timeout.